llvm/lib/Transforms/Scalar/PartiallyInlineLibCalls.cpp

```diff
@@ -56,6 +56,20 @@ static bool optimizeSQRT(CallInst *Call, Function *CalledFunc,
   // dst = phi(v0, v1)
   //
 
+  ORE->emit([&]() {
+    return OptimizationRemark(DEBUG_TYPE, "SqrtPartiallyInlined",
+                              Call->getDebugLoc(), &CurrBB)
+           << "Partially inlined call to sqrt function despite having to use "
+              "errno for error handling: target has fast sqrt instruction";
+  });
+  ORE->emit([&]() {
+    return OptimizationRemarkMissed(DEBUG_TYPE, "BranchInserted",
+                                    Call->getDebugLoc(), &CurrBB)
+           << "Branch to library sqrt fn had to be inserted to satisfy the "
+              "current target's requirement for math functions to set errno on "
+              "invalid inputs";
```

Comment:
Not sure if the `OptimizationRemarkMissed` is the best category for this. Maybe include it in the remark above, as it isn't really a missed optimization `BranchInserted` only happens if`SqrtPartiallyInlined` also happens ?

Or maybe just use `OptimizationRemark`

Reply:
If my explanation is satisfactory, and there are no further concerns to be discussed, may I request a merge?
I not not have write access to the LLVM repo.